hw/battery/src/battery_adc.c

```diff
@@ -104,6 +122,18 @@ battery_adc_open(struct os_dev *dev, uint32_t timeout, void *arg)
         /* Setup channel configuration to use for battery voltage */
         adc_chan_config(bat_adc->adc_dev, bat_adc->cfg.channel,
                 bat_adc->cfg.adc_channel_cfg);
+
+        /* Additional GPIO needed before measurement ? */
+        if (bat_adc->cfg.activation_pin != -1) {
+            /* Should it be activate once or before each measurement */
+            if (bat_adc->cfg.activation_pin_on_open) {
```

Comment:
I have to say I do not understand why you need the activation_pin_on_open flag. Why have the extra logic?